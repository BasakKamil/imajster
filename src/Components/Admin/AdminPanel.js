import React, { Component } from 'react';

class AdminPanel extends Component{
constructor(){
    super();
    this.state = {
        product:{
            name : "",
            category : "",
            description: "",
            OnStock: true,
            image: ""

        }
        // Tablica gdzie będe trzymał produkty -przeniesiona do App
        // products : []
    };


}
handleCHange = (event) => {
    // let newValue = event.target.value;
    // console.log(newValue);
  

    let newproduct;
    if(event.target.name==="OnStock"){
    newproduct = {
        ...this.state.product,
        [event.target.name]: event.target.checked
    }; 
    } else {
        newproduct = {
            ...this.state.product,
            [event.target.name]: event.target.value
        };
    } 
    this.setState({
    product: newproduct
    });
}

addNewProduct= (event) => {
    //Załatwia ze strona nie bedzie sie przeładowywać po nacisnieciu Add 
    event.preventDefault();
    // Tworze kopie swojego stanu by moc ja akytualizowac
    let newProduct = {...this.state.product};
    //Funkcja podpieta pod komponent w APP - przkazywanie props 
    this.props.addProduct(newProduct);

    this.setState({
        product:{
            name : "",
            category : "",
            description: "",
            OnStock: true,
            image: ""
    
        }
    });


}

render(){

return(
<div className="AdminPanelAdd col-xs-2">
    <form onSubmit={this.addNewProduct}>
        <h1>Dodaj Produkt do Bazy Danych</h1>
    <div className="form-group">
        <input type="text" placeholder="name" id="name" name="name"  onChange={this.handleCHange} className="form-control" value={this.state.product.name}/>
    </div>
    <div className="form-group">
        <input type="text" placeholder="category" id="category" name="category" onChange={this.handleCHange} className="form-control" value={this.state.product.category}/>
    </div>
    <div className="form-group">
        <textarea placeholder="description" id="description" name="description" onChange={this.handleCHange} className="form-control" value={this.state.product.description}/>
    </div>
    <div className="form-group">
        <input type="checkbox" placeholder="OnStock" id="OnStock" name="OnStock" onChange={this.handleCHange} className="form-chceck-input" value={this.state.product.OnStock} />
        <label htmlFor="ProductOnStock" className="form-check-label">On Stock</label>
    </div>
    <div className="form-group">
        <input type="text" placeholder="image" id="image" name="image" onChange={this.handleCHange}  className="form-control" value={this.state.product.image} />
    </div>
    <button type="submit" className="btn btn-danger">Add</button>
    </form>



</div>

)

}


}

export default AdminPanel