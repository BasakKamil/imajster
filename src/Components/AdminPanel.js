import React, { Component } from 'react';
import UserForm from './User/UserForm.js';
import LoginPanel from './LoginPanel.js';
import AddBookForm from './AddBookForm.js';
import ShowProducts from './ShowProducts.js';
import fbase from '../base';
import { Link }  from 'react-router-dom';

class AdminPanel extends Component{
constructor(){
    super();
    this.state = {
        // Express Login- backend
        apiResponse : "",
        loggedIn : false ,
        products: [],
        editMode : false,

     
    };


}

  // Express - Tworze backend la React przez API
callAPI(){
    fetch("http://localhost:4000/testAPI")
        .then(res=> res.text())
        .then(res=> this.setState({apiResponse: res}))
        .catch(err => console.log( "Error, probably API serve is disconect..." ));
        
} 




// // Podpiecie Bazy Danych z obecnymi Stanami 
componentDidMount(){
    this.callAPI();
    this.ref = fbase.syncState('store/products',{
        context: this,
        state: 'products'
    });

}

changeLoggedIn = (newValue) => this.setState({loggedIn: newValue}) 

removeFromInventory = (title) => {
    this.setState({
        products: this.state.products.filter(product => title!==product.name)
    });
  
  }

editProduct = () =>{
   this.setState({editMode: true});

  }


render(){
    // const { background = 'yellow' } = this.props;


    // Zamiast  <ShowProducts product={this.state.product}/> moze byc: 
    // const productShow =  this.state.products.map(product => {
    //     return <ShowProducts product={product} />
    // });

return(
<div className="Logowanie">
    <div className="ReturnHome"> <button className="btn btn-danger NowyL" ><Link data-router-view="home" to="/">Powr√≥t</Link></button></div>
    <div className="ReturnHome"> <button className="btn btn-danger NowyL" ><Link data-router-view="cv" to="/cv">CV</Link></button></div>
    <div className="NewLogin">
        <UserForm/>
    </div>

    {!this.state.loggedIn && 
        <LoginPanel
        changeLoggedIn = {this.changeLoggedIn}
        />
    }
    {this.state.loggedIn &&
        <div>
        <AddBookForm 
        changeLoggedIn = {this.changeLoggedIn}
        editMode = {this.state.editMode}
        />
            <div class="Stm">
                <h2>Stan magazynowy:</h2>
                <ShowProducts products={this.state.products} removeFromInventory = {this.removeFromInventory} editProduct={this.editProduct}/>
                {/* {productShow} */}
            </div>
        </div>
    }
    <div className="ApiBackExpress">{this.state.apiResponse}</div>
</div>
)

}


}

export default AdminPanel