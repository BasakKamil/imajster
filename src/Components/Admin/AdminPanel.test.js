import React from 'react';
import ReactDOM from 'react-dom';
import AdminPanel from './AdminPanel.js';

import {configure,shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';



configure({adapter: new Adapter()});

describe('AdminPanel tests',()=>{

    it('AdminPanel render without a problem', ()=>{
        const div = document.createElement('div');
        ReactDOM.render(<AdminPanel/>,div);
        ReactDOM.unmountComponentAtNode(div);
    })
        it('Panel Administrator renders',()=> {
        let wrapper = shallow(<AdminPanel/>);
        expect(wrapper).toMatchSnapshot();
    })
})