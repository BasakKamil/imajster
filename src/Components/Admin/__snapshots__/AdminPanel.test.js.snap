// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminPanel tests Panel Administrator renders 1`] = `ShallowWrapper {}`;
